snuba/web/rpc/v1/resolvers/R_eap_spans/common/sampling_in_storage_util.py

```diff
@@ -1,23 +1,127 @@
+import uuid
+from typing import Callable, TypeVar, cast
+
+from google.protobuf.json_format import MessageToDict
 from sentry_protos.snuba.v1.downsampled_storage_pb2 import DownsampledStorageConfig
 from sentry_protos.snuba.v1.endpoint_time_series_pb2 import TimeSeriesRequest
 from sentry_protos.snuba.v1.endpoint_trace_item_table_pb2 import TraceItemTableRequest
 
+from snuba.attribution import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.datasets.pluggable_dataset import PluggableDataset
 from snuba.downsampled_storage_tiers import Tier
-from snuba.query.query_settings import HTTPQuerySettings
-from snuba.web.rpc.common.debug_info import setup_trace_query_settings
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings, QuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.utils.metrics.timer import Timer
+from snuba.web import QueryResult
+from snuba.web.query import run_query
+
+T = TypeVar("T", TimeSeriesRequest, TraceItemTableRequest)
+
+
+SENTRY_TIMEOUT = 30000  # 30s = 30000ms
+ERROR_BUDGET = 5000  # 5s
```

Comment:
you should make this runtime configurable so you can change it without redeploying